common/flagdproxy/flagdproxy.go

```diff
@@ -62,32 +64,44 @@ func (f *FlagdProxyHandler) Config() *FlagdProxyConfiguration {
 	return f.config
 }
 
+func (f *FlagdProxyHandler) CreateObject(ctx context.Context, obj client.Object) error {
+	return f.Client.Create(ctx, obj)
+}
+
+func (f *FlagdProxyHandler) UpdateObject(ctx context.Context, obj client.Object) error {
+	return f.Client.Update(ctx, obj)
+}
```

Comment:
Both can be non-exported methods. And you may consider inlining these operations